File: databinding/src/main/java/cn/ljpc/databinding/MainActivity2.java
package cn.ljpc.databinding;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import cn.ljpc.databinding.databinding.ActivityMainBinding;

public class MainActivity2 extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.setIdol(new Idol("留洋", "五星"));
        binding.setHander(new EventHandler(this));
        //使用liveData时，需要
        binding.setLifecycleOwner(this);
    }
}

File: mvvm-kotlin/src/main/java/cn/ljpc/view/MainActivity2.kt
package cn.ljpc.view

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import cn.ljpc.R
import cn.ljpc.databinding.ActivityMainBinding
import cn.ljpc.model.Animal
import cn.ljpc.viewmodel.AnimalViewModel

class MainActivity2 : AppCompatActivity() {

    lateinit var mBinding: ActivityMainBinding

    lateinit var mViewModel: AnimalViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        mViewModel = AnimalViewModel(Animal("哈巴狗", 0))
        mBinding.vm = mViewModel
        //LiveData
        mBinding.lifecycleOwner = this
    }
}